frontend/src/metabase-lib/v1/parameters/utils/targets.ts

```diff
@@ -58,13 +58,27 @@ export function getParameterTargetField(
   }
 
   if (isConcreteFieldReference(fieldRef)) {
-    const fieldId = fieldRef[1];
-    const resultMetadata = question.getResultMetadata();
-    const fieldMetadata = resultMetadata.find(field => field.id === fieldId);
-    return (
-      metadata.field(fieldId, fieldMetadata?.table_id) ??
-      metadata.field(fieldId)
+    const query = question.query();
+    const stageIndex = -1;
+    const columns = Lib.visibleColumns(query, stageIndex);
+    const [columnIndex] = Lib.findColumnIndexesFromLegacyRefs(
+      query,
+      stageIndex,
+      columns,
+      [fieldRef],
     );
+    if (columnIndex >= 0) {
```

Comment:
We need to find the corresponding legacy field because:
1) parameters don't use MBQL lib
2) field refs differ between MBQL lib & QP metadata and our existing logic depends on QP metadata